From 867fe44177198fd6b8c0560dee23586300e111b9 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Thu, 13 Feb 2025 00:47:08 +0100
Subject: [PATCH 12/12] Fix spam reload

---
 .../net/citizensnpcs/nms/v1_21_R3/entity/HumanController.java    | 1 +
 1 file changed, 1 insertion(+)

diff --git a/v1_21_R3/src/main/java/net/citizensnpcs/nms/v1_21_R3/entity/HumanController.java b/v1_21_R3/src/main/java/net/citizensnpcs/nms/v1_21_R3/entity/HumanController.java
index aaadbac29..ec7bf5338 100644
--- a/v1_21_R3/src/main/java/net/citizensnpcs/nms/v1_21_R3/entity/HumanController.java
+++ b/v1_21_R3/src/main/java/net/citizensnpcs/nms/v1_21_R3/entity/HumanController.java
@@ -57,6 +57,7 @@ public class HumanController extends AbstractEntityController {
         new fr.euphyllia.utils.FoliaRunnable(fr.euphyllia.utils.FoliaRunnable.GLOBAL()) {
             @Override
             public void run() {
+                if (getBukkitEntity() == null) return;
                 new fr.euphyllia.utils.FoliaRunnable(getBukkitEntity(), null) {
                     @Override
                     public void run() {
-- 
2.47.1.windows.2

