From d0887e0c2c29cb9a3944c73d24451dc588c4355b Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Tue, 12 Aug 2025 11:44:12 +0200
Subject: [PATCH 16/39] Patch trait PausePathfindingTrait.java

---
 .../net/citizensnpcs/trait/PausePathfindingTrait.java  | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/main/src/main/java/net/citizensnpcs/trait/PausePathfindingTrait.java b/main/src/main/java/net/citizensnpcs/trait/PausePathfindingTrait.java
index 4470cfbe6..a3d07f09e 100644
--- a/main/src/main/java/net/citizensnpcs/trait/PausePathfindingTrait.java
+++ b/main/src/main/java/net/citizensnpcs/trait/PausePathfindingTrait.java
@@ -22,7 +22,7 @@ public class PausePathfindingTrait extends Trait {
     @Persist("rightclick")
     private boolean rightclick;
     private int t;
-    private int unpauseTaskId = -1;
+    private net.citizensnpcs.api.util.schedulers.SchedulerTask unpauseTaskId = null;
 
     public PausePathfindingTrait() {
         super("pausepathfinding");
@@ -49,12 +49,12 @@ public class PausePathfindingTrait extends Trait {
     }
 
     private void pause() {
-        if (unpauseTaskId != -1) {
-            Bukkit.getScheduler().cancelTask(unpauseTaskId);
+        if (unpauseTaskId != null) {
+            unpauseTaskId.cancel();
         }
         npc.getNavigator().cancelNavigation();
         npc.getNavigator().setPaused(true);
-        unpauseTaskId = Bukkit.getScheduler().scheduleSyncDelayedTask(CitizensAPI.getPlugin(), () -> {
+        unpauseTaskId = CitizensAPI.getScheduler().runEntityTaskLater(npc.getEntity(), () -> {
             NMS.setPitch(npc.getEntity(), 0);
             npc.getNavigator().setPaused(false);
         }, pauseTicks <= 0 ? 20 : pauseTicks);
@@ -68,7 +68,7 @@ public class PausePathfindingTrait extends Trait {
     @Override
     public void run() {
         if (lockoutDuration > t++ || playerRange == -1 || !npc.isSpawned()
-                || unpauseTaskId == -1 && !npc.getNavigator().isNavigating())
+                || unpauseTaskId == null && !npc.getNavigator().isNavigating())
             return;
 
         if (CitizensAPI.getLocationLookup()
-- 
2.50.1.windows.1

