From ded44b25613902b1ca3952b6311fd7ee591eae3d Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Sun, 10 Aug 2025 15:48:03 +0200
Subject: [PATCH 20/27] Patch trait/waypoint/GuidedWaypointProvider.java

---
 .../trait/waypoint/GuidedWaypointProvider.java            | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/main/src/main/java/net/citizensnpcs/trait/waypoint/GuidedWaypointProvider.java b/main/src/main/java/net/citizensnpcs/trait/waypoint/GuidedWaypointProvider.java
index f296c89f1..4b64cbdc5 100644
--- a/main/src/main/java/net/citizensnpcs/trait/waypoint/GuidedWaypointProvider.java
+++ b/main/src/main/java/net/citizensnpcs/trait/waypoint/GuidedWaypointProvider.java
@@ -145,7 +145,7 @@ public class GuidedWaypointProvider implements EnumerableWaypointProvider {
                     return;
                 if (event.getMessage().equalsIgnoreCase("triggers")) {
                     event.setCancelled(true);
-                    Bukkit.getScheduler().scheduleSyncDelayedTask(CitizensAPI.getPlugin(), () -> {
+                    CitizensAPI.getScheduler().runEntityTask(player, () -> {
                         conversation = TriggerEditPrompt.start(player, this);
                         conversation.addConversationAbandonedListener(e -> {
                             setPaused(false);
@@ -155,10 +155,10 @@ public class GuidedWaypointProvider implements EnumerableWaypointProvider {
                     });
                 } else if (event.getMessage().equalsIgnoreCase("toggle path")) {
                     event.setCancelled(true);
-                    Bukkit.getScheduler().scheduleSyncDelayedTask(CitizensAPI.getPlugin(), this::togglePath);
+                    CitizensAPI.getScheduler().runTask(this::togglePath);
                 } else if (event.getMessage().equalsIgnoreCase("clear")) {
                     event.setCancelled(true);
-                    Bukkit.getScheduler().scheduleSyncDelayedTask(CitizensAPI.getPlugin(), () -> {
+                    CitizensAPI.getScheduler().runTask(() -> {
                         destinations.clear();
                         guides.clear();
                         if (showPath) {
@@ -170,7 +170,7 @@ public class GuidedWaypointProvider implements EnumerableWaypointProvider {
                     double d = Double.parseDouble(event.getMessage().replace("distance ", "").trim());
                     if (d <= 0)
                         return;
-                    Bukkit.getScheduler().scheduleSyncDelayedTask(CitizensAPI.getPlugin(), () -> {
+                    CitizensAPI.getScheduler().runTask(() -> {
                         distance = (float) d;
                         Messaging.sendTr(sender, Messages.GUIDED_WAYPOINT_EDITOR_DISTANCE_SET, d);
                     });
-- 
2.50.1.windows.1

