From 5084271f08d7b19cd3720264fe57c64fdc634e46 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Tue, 12 Aug 2025 11:44:13 +0200
Subject: [PATCH 21/39] Patch trait waypoint LinearWaypointProvider.java

---
 .../trait/waypoint/LinearWaypointProvider.java         | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/main/src/main/java/net/citizensnpcs/trait/waypoint/LinearWaypointProvider.java b/main/src/main/java/net/citizensnpcs/trait/waypoint/LinearWaypointProvider.java
index 4e7291871..610234269 100644
--- a/main/src/main/java/net/citizensnpcs/trait/waypoint/LinearWaypointProvider.java
+++ b/main/src/main/java/net/citizensnpcs/trait/waypoint/LinearWaypointProvider.java
@@ -359,7 +359,7 @@ public class LinearWaypointProvider implements EnumerableWaypointProvider {
             String message = event.getMessage();
             if (message.equalsIgnoreCase("triggers")) {
                 event.setCancelled(true);
-                Bukkit.getScheduler().scheduleSyncDelayedTask(CitizensAPI.getPlugin(), () -> {
+                CitizensAPI.getScheduler().runEntityTask(event.getPlayer(), () -> {
                     conversation = TriggerEditPrompt.start(player, LinearWaypointEditor.this);
                     conversation.addConversationAbandonedListener(e -> {
                         setPaused(false);
@@ -369,20 +369,20 @@ public class LinearWaypointProvider implements EnumerableWaypointProvider {
                 });
             } else if (message.equalsIgnoreCase("clear")) {
                 event.setCancelled(true);
-                Bukkit.getScheduler().scheduleSyncDelayedTask(CitizensAPI.getPlugin(), this::clearWaypoints);
+                CitizensAPI.getScheduler().runEntityTask(event.getPlayer(), this::clearWaypoints);
             } else if (message.equalsIgnoreCase("toggle path") || message.equalsIgnoreCase("markers")) {
                 event.setCancelled(true);
-                Bukkit.getScheduler().scheduleSyncDelayedTask(CitizensAPI.getPlugin(), this::togglePath);
+                CitizensAPI.getScheduler().runEntityTask(event.getPlayer(), this::togglePath);
             } else if (message.equalsIgnoreCase("cycle")) {
                 event.setCancelled(true);
-                Bukkit.getScheduler().scheduleSyncDelayedTask(CitizensAPI.getPlugin(), () -> {
+                CitizensAPI.getScheduler().runEntityTask(event.getPlayer(), () -> {
                     cycle = !cycle;
                     Messaging.sendTr(event.getPlayer(), cycle ? Messages.LINEAR_WAYPOINT_EDITOR_CYCLE_SET
                             : Messages.LINEAR_WAYPOINT_EDITOR_CYCLE_UNSET);
                 });
             } else if (message.equalsIgnoreCase("here")) {
                 event.setCancelled(true);
-                Bukkit.getScheduler().scheduleSyncDelayedTask(CitizensAPI.getPlugin(),
+                CitizensAPI.getScheduler().runEntityTask(event.getPlayer(),
                         () -> addWaypoint(player.getLocation()));
             }
         }
-- 
2.50.1.windows.1

