From bd7672bf0256d398fe9123bac62d747da1aeaec2 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Tue, 12 Aug 2025 11:44:18 +0200
Subject: [PATCH 36/39] Patch nms v1_20_R4 util NMSImpl.java

---
 .../nms/v1_20_R4/util/NMSImpl.java            | 23 +++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/NMSImpl.java b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/NMSImpl.java
index 128551cac..954701213 100644
--- a/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/NMSImpl.java
+++ b/v1_20_R4/src/main/java/net/citizensnpcs/nms/v1_20_R4/util/NMSImpl.java
@@ -410,6 +410,29 @@ public class NMSImpl implements NMSBridge {
         return success;
     }
 
+    public void addEntityToWorld(org.bukkit.entity.Entity entity, SpawnReason custom, java.util.function.Consumer<Boolean> isAdded) {
+        if (!net.citizensnpcs.api.util.SpigotUtil.isFoliaServer()) {
+            isAdded.accept(addEntityToWorld(entity, custom));
+            return;
+        }
+        int viewDistance = -1;
+        ChunkMap chunkMap = null;
+        try {
+            if (entity instanceof Player) {
+                chunkMap = ((ServerChunkCache) getHandle(entity).level().getChunkSource()).chunkMap;
+                viewDistance = chunkMap.serverViewDistance;
+                chunkMap.serverViewDistance = -1;
+            }
+        } catch (Throwable e) {
+            e.printStackTrace();
+        }
+        boolean success = getHandle(entity).level().addFreshEntity(getHandle(entity), custom);
+        if (chunkMap != null) {
+            chunkMap.serverViewDistance = viewDistance;
+        }
+        isAdded.accept(success);
+    }
+
     @Override
     public void addOrRemoveFromPlayerList(org.bukkit.entity.Entity entity, boolean remove) {
         if (entity == null)
-- 
2.50.1.windows.1

