From 77b7cc49c3e2a4da6d046a7aacc95b60cfed218c Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Tue, 12 Aug 2025 11:26:37 +0200
Subject: [PATCH 37/39] Patch nms v1_21_R5 entity HumanController.java

---
 .../citizensnpcs/nms/v1_21_R5/entity/HumanController.java    | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/v1_21_R5/src/main/java/net/citizensnpcs/nms/v1_21_R5/entity/HumanController.java b/v1_21_R5/src/main/java/net/citizensnpcs/nms/v1_21_R5/entity/HumanController.java
index b70d491b8..5e0dab722 100644
--- a/v1_21_R5/src/main/java/net/citizensnpcs/nms/v1_21_R5/entity/HumanController.java
+++ b/v1_21_R5/src/main/java/net/citizensnpcs/nms/v1_21_R5/entity/HumanController.java
@@ -54,11 +54,12 @@ public class HumanController extends AbstractEntityController {
                 e.printStackTrace();
             }
         }
-        Bukkit.getScheduler().scheduleSyncDelayedTask(CitizensAPI.getPlugin(), () -> {
+        CitizensAPI.getScheduler().runTaskLater(() -> {
+            if (getBukkitEntity() == null) return;
             if (getBukkitEntity() == null || !getBukkitEntity().isValid()
                     || getBukkitEntity() != handle.getBukkitEntity())
                 return;
-            NMS.addOrRemoveFromPlayerList(getBukkitEntity(), npc.shouldRemoveFromPlayerList());
+            CitizensAPI.getScheduler().runEntityTask(getBukkitEntity(), () -> NMS.addOrRemoveFromPlayerList(getBukkitEntity(), npc.shouldRemoveFromPlayerList()));
         }, 20);
         handle.getBukkitEntity().setSleepingIgnored(true);
         return handle.getBukkitEntity();
-- 
2.50.1.windows.1

