From f4f9d0c1bd8348ab80f08456fb4830de42320aeb Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Tue, 12 Aug 2025 11:26:37 +0200
Subject: [PATCH 38/39] Patch nms v1_21_R5 util CitizensEntityTracker.java

---
 .../v1_21_R5/util/CitizensEntityTracker.java  | 59 +++++++++++--------
 1 file changed, 34 insertions(+), 25 deletions(-)

diff --git a/v1_21_R5/src/main/java/net/citizensnpcs/nms/v1_21_R5/util/CitizensEntityTracker.java b/v1_21_R5/src/main/java/net/citizensnpcs/nms/v1_21_R5/util/CitizensEntityTracker.java
index 3d16b0b17..97da5e1f1 100644
--- a/v1_21_R5/src/main/java/net/citizensnpcs/nms/v1_21_R5/util/CitizensEntityTracker.java
+++ b/v1_21_R5/src/main/java/net/citizensnpcs/nms/v1_21_R5/util/CitizensEntityTracker.java
@@ -77,34 +77,43 @@ public class CitizensEntityTracker extends ChunkMap.TrackedEntity {
             if (REQUIRES_SYNC == null) {
                 REQUIRES_SYNC = !Bukkit.isPrimaryThread();
             }
-            boolean cancelled = Util.callPossiblySync(() -> {
-                NPCSeenByPlayerEvent event = new NPCSeenByPlayerEvent(npc, entityplayer.getBukkitEntity());
-                try {
-                    Bukkit.getPluginManager().callEvent(event);
-                } catch (IllegalStateException e) {
-                    REQUIRES_SYNC = true;
-                    throw e;
-                }
-                if (event.isCancelled())
-                    return true;
-
-                Integer trackingRange = npc.data().<Integer> get(NPC.Metadata.TRACKING_RANGE);
-                if (TRACKING_RANGE_SETTER != null && trackingRange != null
-                        && npc.data().get("last-tracking-range", -1) != trackingRange.intValue()) {
-                    try {
-                        TRACKING_RANGE_SETTER.invoke(CitizensEntityTracker.this, trackingRange);
-                        npc.data().set("last-tracking-range", trackingRange);
-                    } catch (Throwable e) {
-                        e.printStackTrace();
-                    }
+            cancelledUpdatePlayer(npc, entityplayer, cancelled -> {
+                if (cancelled) {
+                    return;
                 }
-                return false;
-            }, REQUIRES_SYNC);
+                super.updatePlayer(entityplayer);
+            });
+        } else {
+            super.updatePlayer(entityplayer);
+        }
+    }
 
-            if (cancelled)
+    private void cancelledUpdatePlayer(final NPC npc, final ServerPlayer entityplayer, java.util.function.Consumer<Boolean> callback) {
+        net.citizensnpcs.api.CitizensAPI.getScheduler().runEntityTask(entityplayer.getBukkitEntity(), () -> {
+            NPCSeenByPlayerEvent event = new NPCSeenByPlayerEvent(npc, entityplayer.getBukkitEntity());
+            try {
+                Bukkit.getPluginManager().callEvent(event);
+            } catch (IllegalStateException e) {
+                REQUIRES_SYNC = true;
+                throw e;
+            }
+            if (event.isCancelled()) {
+                callback.accept(true);
                 return;
-        }
-        super.updatePlayer(entityplayer);
+            }
+
+            Integer trackingRange = npc.data().<Integer> get(NPC.Metadata.TRACKING_RANGE);
+            if (TRACKING_RANGE_SETTER != null && trackingRange != null
+                    && npc.data().get("last-tracking-range", -1) != trackingRange.intValue()) {
+                try {
+                    TRACKING_RANGE_SETTER.invoke(CitizensEntityTracker.this, trackingRange);
+                    npc.data().set("last-tracking-range", trackingRange);
+                } catch (Throwable e) {
+                    e.printStackTrace();
+                }
+            }
+            callback.accept(false);
+        });
     }
 
     public static Collection<org.bukkit.entity.Entity> getSeenBy(TrackedEntity tracker) {
-- 
2.50.1.windows.1

