From b41aa9056dfd1648a3473afde686be9379d5c906 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Tue, 12 Aug 2025 11:44:19 +0200
Subject: [PATCH 39/39] Patch nms v1_21_R5 util NMSImpl.java

---
 .../nms/v1_21_R5/util/NMSImpl.java            | 104 +++++++++++++++---
 1 file changed, 90 insertions(+), 14 deletions(-)

diff --git a/v1_21_R5/src/main/java/net/citizensnpcs/nms/v1_21_R5/util/NMSImpl.java b/v1_21_R5/src/main/java/net/citizensnpcs/nms/v1_21_R5/util/NMSImpl.java
index 289dd88ca..7a24d2c16 100644
--- a/v1_21_R5/src/main/java/net/citizensnpcs/nms/v1_21_R5/util/NMSImpl.java
+++ b/v1_21_R5/src/main/java/net/citizensnpcs/nms/v1_21_R5/util/NMSImpl.java
@@ -408,6 +408,9 @@ public class NMSImpl implements NMSBridge {
 
     @Override
     public boolean addEntityToWorld(org.bukkit.entity.Entity entity, SpawnReason custom) {
+        if (net.citizensnpcs.api.util.SpigotUtil.isFoliaServer()) {
+            throw new UnsupportedOperationException("NMSImpl#addEntityToWorld is not supported on Folia servers.");
+        }
         int viewDistance = -1;
         ChunkMap chunkMap = null;
 
@@ -423,6 +426,31 @@ public class NMSImpl implements NMSBridge {
         return success;
     }
 
+    public void addEntityToWorld(org.bukkit.entity.Entity entity, SpawnReason custom, java.util.function.Consumer<Boolean> isAdded) {
+        if (!net.citizensnpcs.api.util.SpigotUtil.isFoliaServer()) {
+            isAdded.accept(addEntityToWorld(entity, custom));
+            return;
+        }
+        final ServerLevel serverLevel = (ServerLevel) getHandle(entity).level();
+        int viewDistance = -1;
+        ChunkMap chunkMap = null;
+        if (entity instanceof Player) {
+            chunkMap = ((ServerChunkCache) serverLevel.getChunkSource()).chunkMap;
+            viewDistance = chunkMap.serverViewDistance;
+            chunkMap.serverViewDistance = -1;
+        }
+        final ChunkMap chunkMapFinal = chunkMap;
+        final int viewDistanceFinal = viewDistance;
+        final Location location = entity.getLocation();
+        CitizensAPI.getScheduler().runRegionTask(location.getWorld(), location.getBlockX() >> 4, location.getBlockZ() >> 4, () -> {
+            boolean success = serverLevel.addFreshEntity(getHandle(entity), custom);
+            if (chunkMapFinal != null) {
+                chunkMapFinal.serverViewDistance = viewDistanceFinal;
+            }
+            isAdded.accept(success);
+        });
+    }
+
     @Override
     public void addOrRemoveFromPlayerList(org.bukkit.entity.Entity entity, boolean remove) {
         if (entity == null)
@@ -1425,20 +1453,37 @@ public class NMSImpl implements NMSBridge {
     @Override
     public void replaceTrackerEntry(org.bukkit.entity.Entity entity) {
         Entity handle = getHandle(entity);
+        TrackedEntity entry;
         ChunkMap cm = ((ServerLevel) handle.level()).getChunkSource().chunkMap;
-        TrackedEntity entry = cm.entityMap.get(entity.getEntityId());
-        if (entry == null)
-            return;
-        entry.broadcastRemoved();
-        CitizensEntityTracker newTracker = new CitizensEntityTracker(cm, entry);
-        for (MethodHandle setter : TRACKED_ENTITY_SETTERS) {
+        CitizensEntityTracker newTracker;
+        if (net.citizensnpcs.api.util.SpigotUtil.isFoliaServer()) {
+            ServerLevel server = (ServerLevel) getHandle(entity).level();
             try {
-                setter.invoke(handle, newTracker);
-            } catch (Throwable e) {
-                e.printStackTrace();
+                entry = getTrackedEntityFolia(handle);
+                if (entry == null) return;
+                entry.broadcastRemoved();
+                newTracker = new CitizensEntityTracker(server.getChunkSource().chunkMap, entry);
+                setTrackedEntityFolia(handle, newTracker);
+            } catch (Exception exception) {
+                CitizensAPI.getPlugin().getLogger().severe("Failed to replace tracker entry for " + entity.getType());
+                exception.printStackTrace(System.err);
+                return;
+            }
+        } else {
+            entry = cm.entityMap.get(entity.getEntityId());
+            if (entry == null)
+                return;
+            entry.broadcastRemoved();
+            newTracker = new CitizensEntityTracker(cm, entry);
+            for (MethodHandle setter : TRACKED_ENTITY_SETTERS) {
+                try {
+                    setter.invoke(handle, newTracker);
+                } catch (Throwable e) {
+                    e.printStackTrace();
+                }
             }
+            cm.entityMap.put(entity.getEntityId(), newTracker);
         }
-        cm.entityMap.put(entity.getEntityId(), newTracker);
     }
 
     @Override
@@ -1627,8 +1672,13 @@ public class NMSImpl implements NMSBridge {
 
     @Override
     public void setLocationDirectly(org.bukkit.entity.Entity entity, Location location) {
-        getHandle(entity).snapTo(location.getX(), location.getY(), location.getZ(), location.getYaw(),
-                location.getPitch());
+        // Todo - A fixer plus tard, problème avec moonrise
+        if (net.citizensnpcs.api.util.SpigotUtil.isFoliaServer()) {
+            net.citizensnpcs.api.util.SpigotUtil.teleportAsync(entity, location);
+        } else {
+            getHandle(entity).snapTo(location.getX(), location.getY(), location.getZ(), location.getYaw(),
+                    location.getPitch());
+        }
     }
 
     @Override
@@ -2312,8 +2362,13 @@ public class NMSImpl implements NMSBridge {
         }
         List<Packet<?>> toSend = Lists.newArrayList();
         if (position) {
-            TrackedEntity entry = ((ServerLevel) handle.level()).getChunkSource().chunkMap.entityMap
-                    .get(handle.getId());
+            TrackedEntity entry = null;
+            if (net.citizensnpcs.api.util.SpigotUtil.isFoliaServer()) {
+                entry = getTrackedEntityFolia(handle);
+            } else {
+                entry = ((ServerLevel) handle.level()).getChunkSource().chunkMap.entityMap
+                        .get(handle.getId());
+            }
             if (entry == null) {
                 Messaging.debug("Null tracker entity for ", from);
                 return Collections.emptyList();
@@ -2840,4 +2895,25 @@ public class NMSImpl implements NMSBridge {
             Messaging.logTr(Messages.ERROR_GETTING_ID_MAPPING, e.getMessage());
         }
     }
+
+    private static TrackedEntity getTrackedEntityFolia(Entity entity) {
+        try {
+            java.lang.reflect.Field field = Entity.class.getDeclaredField("trackedEntity");
+            field.setAccessible(true);
+            return (TrackedEntity) field.get(entity);
+        } catch (Exception exception) {
+            exception.printStackTrace(System.err);
+            return null;
+        }
+    }
+
+    private void setTrackedEntityFolia(Entity entity, TrackedEntity trackedEntity) {
+        try {
+            java.lang.reflect.Field field = Entity.class.getDeclaredField("trackedEntity");
+            field.setAccessible(true);
+            field.set(entity, trackedEntity);
+        } catch (Exception exception) {
+            exception.printStackTrace(System.err);
+        }
+    }
 }
-- 
2.50.1.windows.1

