From 73548bcca8893ceaa3e59aaad56bc802fb450e6a Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Sat, 22 Mar 2025 02:08:34 +0100
Subject: [PATCH 12/15] Fix spam reload

---
 .../net/citizensnpcs/nms/v1_21_R3/entity/HumanController.java    | 1 +
 1 file changed, 1 insertion(+)

diff --git a/v1_21_R3/src/main/java/net/citizensnpcs/nms/v1_21_R3/entity/HumanController.java b/v1_21_R3/src/main/java/net/citizensnpcs/nms/v1_21_R3/entity/HumanController.java
index aaadbac29..ec7bf5338 100644
--- a/v1_21_R3/src/main/java/net/citizensnpcs/nms/v1_21_R3/entity/HumanController.java
+++ b/v1_21_R3/src/main/java/net/citizensnpcs/nms/v1_21_R3/entity/HumanController.java
@@ -57,6 +57,7 @@ public class HumanController extends AbstractEntityController {
         new fr.euphyllia.utils.FoliaRunnable(fr.euphyllia.utils.FoliaRunnable.GLOBAL()) {
             @Override
             public void run() {
+                if (getBukkitEntity() == null) return;
                 new fr.euphyllia.utils.FoliaRunnable(getBukkitEntity(), null) {
                     @Override
                     public void run() {
-- 
2.49.0.windows.1

