From 6c016a7fee9a5a0f6c594febfcb67377657a536c Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Thu, 13 Feb 2025 03:42:01 +0100
Subject: [PATCH 6/7] Add more support in FoliaRunnable

---
 .../fr/euphyllia/utils/FoliaRunnable.java     | 26 +++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/src/main/java/fr/euphyllia/utils/FoliaRunnable.java b/src/main/java/fr/euphyllia/utils/FoliaRunnable.java
index 85dd97e..26a65b8 100644
--- a/src/main/java/fr/euphyllia/utils/FoliaRunnable.java
+++ b/src/main/java/fr/euphyllia/utils/FoliaRunnable.java
@@ -1,8 +1,10 @@
 package fr.euphyllia.utils;
 
 import io.papermc.paper.threadedregions.scheduler.*;
+import org.bukkit.Bukkit;
 import org.bukkit.Location;
 import org.bukkit.World;
+import org.bukkit.entity.Entity;
 import org.bukkit.plugin.Plugin;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
@@ -67,6 +69,18 @@ public abstract class FoliaRunnable implements Runnable {
         this.entityRetired = retired;
     }
 
+    /**
+     * Constructs a new {@code FoliaRunnable} that will be executed on an entity's region.
+     * An optional {@code retired} callback can be provided if the entity is removed before execution.
+     *
+     * @param entity an {@link Entity} to schedule the task on an entity's region
+     * @param retired   a callback invoked if the entity is removed before the task can run
+     */
+    public FoliaRunnable(@NotNull Entity entity, @Nullable Runnable retired) {
+        this.entityScheduler = entity.getScheduler();
+        this.entityRetired = retired;
+    }
+
     /**
      * Constructs a new {@code FoliaRunnable} that will be executed on the global region.
      *
@@ -300,4 +314,16 @@ public abstract class FoliaRunnable implements Runnable {
         this.task = task;
         return task;
     }
+
+    public static GlobalRegionScheduler GLOBAL() {
+        return Bukkit.getGlobalRegionScheduler();
+    }
+
+    public static AsyncScheduler ASYNC() {
+        return Bukkit.getAsyncScheduler();
+    }
+
+    public static RegionScheduler REGION() {
+        return Bukkit.getRegionScheduler();
+    }
 }
\ No newline at end of file
-- 
2.48.1.windows.1

